### PR TITLE
Add datastore acceptance tests

### PR DESCRIPTION
**Pull request by stuart-mclaren-hpe**
The tests can be run against a real system with:

```
$ make testacc
```

or in 'simulation' mode with:

```
$ make testacc-simulation
```

In simulation mode fake http responses are injected using gock. This 
allows running the acceptance tests without requiring real hardware.

### CODE REVIEW

File: internal/simulator/datastore_create.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	gock.New("http://localhost").
+		Get("/data-services/v1beta1/async-operations/"+taskID).
+		Reply(200).
+		SetHeader("Content-Type", "application/json").
+		BodyString(dsAsync5)
+
+	gock.New("http://localhost").
+		Get("/virtualization/v1beta1/datastores").
+		// MatchParam("filter", "name eq "+datastoreName).
```
Comment (ZoeySimone) [inline internal/simulator/datastore_create.go:74]: Stray comment?

File: internal/simulator/datastore_create.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+var dsAsync4 string
+
+//go:embed fixtures/datastores/create/async5.json
+var dsAsync5 string
+
+//go:embed fixtures/datastores/create/filter.json
+var dsFilter string
+
+//go:embed fixtures/datastores/create/get.json
+var dsDet string
```
Comment (ZoeySimone) [inline internal/simulator/datastore_create.go:30]: `dsDet` instead of `dsGet` typo?

Comment (stuart-mclaren-hpe) [reply internal/simulator/datastore_create.go:30]: Yes, thanks. Curious that this was consistent on line 84.

File: test/datastore/datastore_test.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+		checks = append(checks,
+			resource.TestCheckResourceAttr(
+				"hpegl_pc_datastore.test",
+				"id",
+				"698de955-87b5-5fe6-b683-78c3948beede",
+			),
+		)
+	} else {
+		// In non-simulation mode check that the ID is a valid UUID
+		checks = append(
```
[the remark this answers is not in the document]
Comment (stuart-mclaren-hpe) [reply test/datastore/datastore_test.go:107]: I was thinking that the known id is a superset of the is uuid check, so in terms of functionality it doesn't add much.

What you're saying is a minor simplification of the code. This pattern is going to be repeated in follow on tests...I'll make the change.

**Review by ZoeySimone — APPROVED** [review]

**Pull request merged**